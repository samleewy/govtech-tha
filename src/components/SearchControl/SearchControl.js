import React from 'react'
import { getResults } from '../../utils/data'
import SearchButton from '../SearchButton/SearchButton'
import SearchTextBox from '../SearchTextBox/SearchTextBox'

const SearchControl = ({ searchQuery, setSearchQuery, setResults, setLoading, setError }) => {

    const onSearchClick = async () => {
        if (searchQuery.length > 0) {
            setError(false)
            setLoading(true)
            try {
                const data = await getResults()
                setLoading(false)
                setResults(data)
            } catch (err) {
                console.log(err)
                setError(true)
            }
        }
    }

    return (
        <>
            <div style={{ width: '100%', height: 0 }}>
                <SearchTextBox 
                    searchQuery={searchQuery}
                    setSearchQuery={setSearchQuery}
                    onSearchClick={onSearchClick} />
            </div>
            <div style={{ position: 'relative', left: -10 }}>
                <SearchButton 
                    searchQuery={searchQuery}
                    onSearchClick={onSearchClick} />
            </div>
        </>
    )
}

export default SearchControl