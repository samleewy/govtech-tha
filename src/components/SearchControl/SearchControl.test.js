import { render, fireEvent } from '@testing-library/react';
import SearchControl from './SearchControl'

describe("This suit is to test SearchControl component", () => {
    it('should render correctly', () => {
        const { asFragment } = render(<SearchControl searchQuery=""/>)
        expect(asFragment()).toMatchSnapshot();
    })
    it('should change input value according to given searchQuery prop', () => {
        const testInput = 'child'
        const { getByTestId } = render(<SearchControl searchQuery={testInput} />)
        const searchTextBox = getByTestId('search-textbox')
        expect(searchTextBox.value).toBe(testInput)
    })
})