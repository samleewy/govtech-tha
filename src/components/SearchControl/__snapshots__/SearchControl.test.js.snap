// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`This suit is to test SearchControl component should render correctly 1`] = `
<DocumentFragment>
  <div
    style="width: 100%; height: 0px;"
  >
    <div
      aria-expanded="false"
      aria-haspopup="listbox"
      aria-owns="react-autowhatever-1"
      class="react-autosuggest__container"
      data-testid="search-textbox-suggest"
      role="combobox"
    >
      <input
        aria-autocomplete="list"
        aria-controls="react-autowhatever-1"
        autocomplete="off"
        class="react-autosuggest__input react-autosuggest__input--focused"
        data-testid="search-textbox"
        type="text"
        value=""
      />
      <div
        class="react-autosuggest__suggestions-container"
        id="react-autowhatever-1"
        role="listbox"
      />
    </div>
  </div>
  <div
    style="position: relative; left: -10px;"
  >
    <button>
      <i
        class="fas fa-search"
      />
      <span
        class="search-btn-text"
      >
        Search
      </span>
    </button>
  </div>
</DocumentFragment>
`;
